app/policies/project_policy.rb

```diff
@@ -0,0 +1,36 @@
+class ProjectPolicy < ApplicationPolicy
+  def editor?
+    return false if records.empty?
+    records.compact.all? do |record|
+      editor_project_ids.include? record.id.to_s
```

Comment:
See other comment about using class methods for testing the user roles for this project record.

Also It would read better if this had a method wrapping it, perhaps, `can_edit_project?(record.id)` and i'd encapsulate the `.to_s` to that method.

Same for the methods below.

Reply:
The "can edit" question (not projects, but transcriptions via project roles) is what the `editor?` predicate method is for. You think this needs to be broken up further?

Reply:
I think you can extract the common behaviour for ensuring all records have the expected record id in it, looks very similar across the methods here, what you call that method is up to you.

Reply:
👍